addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1253,6 +1252,11 @@ var SnippetsMenu = Widget.extend({
                 // tools is only useful for out-of-snippet texts anyway, so
                 // snippet texts can still be enabled immediately.
                 this._mutex.exec(() => this._textToolsSwitchingEnabled = true);
+                // We updateInvisibleDOM as late as possible because in the
+                // start of VisibilityPageOptionUpdate, the option trigger an
+                // "action_demand" which is managed by the navbar. We do this as
+                // late as possible so that the navbar had the time to launch.
+                this._updateInvisibleDOM();
```

Comment:
I don't understand how that analyze can be right: this widget here (`SnippetMenu`) is created by the widget in charge of starting the edit mode somehow (`EditPageMenu`) which is itself created by the navbar (as declared with `websiteNavbarData.websiteNavbarRegistry.add(EditPageMenu, '#edit-page-menu');`)... and even if an `action_demand` is received by the navbar "too soon", there is some protection:
```javascript
_handleAction: function (actionName, params, _i) {
        var self = this;
        return this._whenReadyForActions().then(function () {
```
(`_whenReadyForActions()` being resolved once all subwidgets are properly started). What's the result of your investigation around all of this?

Reply:
@qsm-odoo This is what I understood:
To be "Ready for actions" the `WebsiteNavbar` wait for `EditPageMenu` to be fully started.
The `EditPageMenu` wait for the EditorMenu
Then it passes to few component to arrive to `SnippetsMenu` which await `_updateInvisibleDOM`
The `updateInvisbileDom` function creates snippets editors for invisible snippets which wait for the options to start.
Finally the option `VisibilityPageOptionUpdate` await the result of action_demand which is blocked by the navbar.

This is where the snake bites its tail. The navbar waits for the invisible snippets options to be started so that it is ready for actions and the VisibilityPageOptionUpdate waits for the navbar to be ready to perform actions to start.

This is visible in the stack trace when you toggle off the footer visibility option, save the page and then try to go `/?enable_editor=1` with a debugger in the start of `VisibilityPageOptionUpdate`.

Reply:
Maybe a better solution would be to change this
```
const shown = await this._isShown();
this.trigger_up('snippet_option_visibility_update', {show: shown});
```
to
```
this._isShown().then(isShown => {
       this.trigger_up('snippet_option_visibility_update', { show: isShown })
 });
```

Reply:
Seems like a good investigation and good solution :+1:
- Your investigation seems to be the right one: you are explaining a **forced deadlock** (it always happens if you follow that flow, right?) => it's quite the opposite from a "race condition" like you speak about in your commit message (a race condition is something which happens sometimes, if it's a forced deadlock, it's not a race condition).
- Your solution seems to be the right one: while in general it's better for each caller to properly await the async parts it triggers, in this case I don't think the option needs to include its `snippet_option_visibility_update` before saying it is ready. It's quite a special case for an option to have to trigger `snippet_option_visibility_update` in its `start`, in the end it should not be necessary: it's probably just an error we can fix in master: there should be `data-invisible="1"` on the footer when it's hidden for the page, that way a call to 'snippet_option_visibility_update' in the start would always be useless (just like popup: they have data-invisible="1" saved with the page).

So do that, avoid speaking about race conditions but explain your forced deadlock and add comments next to the related code, explaining why and how it could be refactored in master (with a TODO) (check if what I explain to be done for master works first (adding data-invisible="1" in the template rendering when the footer is hidden (which we cannot do in stable but you can test it)).